source/Politics.cpp

```diff
@@ -171,7 +171,8 @@ bool Politics::CanLand(const Ship &ship, const Planet *planet) const
 
 	const Government *gov = ship.GetGovernment();
 	if(!gov->IsPlayer())
-		return !IsEnemy(gov, planet->GetGovernment());
+		return !IsEnemy(gov, planet->GetGovernment()) &&
+			(ship.GetPersonality().IsUnrestricted() || !gov->IsRestrictedFrom(*planet));
```

Comment:
Note that before getting here, we're returning true if the planet is uninhabited. This means that a ship can never be restricted from landing on an uninhabited planet.

The `return CanLand(planet);` below this is only used for the player and already contains its own `if(!planet->IsInhabited())` check. Therefore, we could move the above uninhabited check into here. We'd then check "not enemy and (not restricted or not inhabited)".

Reply:
alr I did it but I'm confused as to why you'd be restricted from landing on an uninhabited enemy planet

Reply:
right now Shield Beetles are allowed to land on the uninhabited Hai stations which is a bit annoying so maaaybe this is a bit of a bugfix

Reply:
>did you mean I should do "not restricted and (not inhabited or not enemy)"?

That'd be better.